app/scripts/directives/logViewer.js

```diff
@@ -310,117 +318,124 @@ angular.module('openshiftConsole')
                 cachedLogNode && (cachedLogNode.innerHTML = '');
                 buffer = document.createDocumentFragment();
               }
+
+              return deferred.promise;
             };
 
             var streamLogs = function() {
               // Stop any active streamer.
-              stopStreaming();
-
-              if(!$scope.run) {
-                return;
-              }
-
-              angular.extend($scope, {
-                loading: true,
-                autoScrollActive: true,
-                limitReached: false,
-                showScrollLinks: false
-              });
+              stopStreaming().then(function() {
+                $scope.$evalAsync(function() {
+                  if(!$scope.run) {
+                    return;
+                  }
 
-              var options = angular.extend({
-                follow: true,
-                tailLines: 5000,
-                // Limit log size to 10 MiB. Note: This can't be more than 500 MiB,
-                // otherwise save log will break because we'll exceed the max Blob
-                // size for some browsers.
-                // https://github.com/eligrey/FileSaver.js#supported-browsers
-                limitBytes: 10 * 1024 * 1024
-              }, $scope.options);
-
-              streamer = DataService.createStream(logSubresource, name, $scope.context, options);
-
-              var lastLineNumber = 0;
-              var addLine = function(text) {
-                lastLineNumber++;
-                // Append the line to the document fragment buffer.
-                buffer.appendChild(buildLogLineNode(lastLineNumber, text));
-                update();
-              };
+                  angular.extend($scope, {
+                    loading: true,
+                    autoScrollActive: true,
+                    largeLog: false,
+                    limitReached: false,
+                    showScrollLinks: false
+                  });
 
-              streamer.onMessage(function(msg, raw, cumulativeBytes) {
-                // ensures the digest loop will catch the state change.
-                $scope.$evalAsync(function() {
-                  $scope.empty = false;
-                  if($scope.state !== 'logs') {
+                  // Prevent flicker switching from a container with an empty log
+                  // to one with content.
+                  if ($scope.state === 'empty') {
```

Comment:
It gets set to empty again here:

https://github.com/spadgett/origin-web-console/blob/b414f5a17ba32e6635cd6cc991be36504fc4368b/app/scripts/directives/logViewer.js#L411